### ISSUE TITLE
invalid JS included via vendor.js blows up `ember test` 

### ISSUE THREAD

**Issue by efx** [ember-cli/ember-cli#6049]
A dependency in my ember-cli project contained invalid JavaScript. When I ran `ember test`, the test process did not start properly and then a number of syntax errors were printed to my terminal. (The test command is just launching phantom)
Here is a repo that demonstrates the issue: 
https://github.com/efx/poc-bug
I think it would be useful to describe the error in a more helpful way.
Shouldn't  [`testem.startCI`](https://github.com/ember-cli/ember-cli/blob/master/lib/tasks/test.js#L18-L19) inform the user when a syntax error early on in `vendor.js` prevents the test suite  from initializing properly? (i.e. like 'No test runner found' error)

related issue: https://github.com/ariya/phantomjs/issues/14173

---

Output from ember version --verbose:
```
ember-cli: 2.7.0-beta.5-master-4ad02d31ca
http_parser: 2.7.0
node: 6.2.1
v8: 5.0.71.52
uv: 1.9.1
zlib: 1.2.8
ares: 1.10.1-DEV
icu: 57.1
modules: 48
openssl: 1.0.2h
os: darwin x64
```

**Comment by nathanhammond** [comment]
@efx Unfortunately what you're seeing is expected behavior. You attempted to run your tests and the first thing that it spits out is:

```
not ok 1 PhantomJS 2.1 - Global error: SyntaxError: Unexpected EOF at http://localhost:7357/assets/vendor.js, line 64376
    ---
        Log: |
            { type: 'error',
              text: 'SyntaxError: Unexpected EOF at http://localhost:7357/assets/vendor.js, line 64376\n' }
    ...
```

... which successfully identifies what the problem is. In this scenario `ember test` works perfectly, and the child components do as well:
- Phantom throws an error when it encounters something it can't handle.
- Testem captures the error and logs it.
- Ember CLI reports on the error via Testem.

Your linked Phantom issue feels [successfully diagnosed](https://github.com/ariya/phantomjs/issues/14173#issuecomment-212663559) by @wagenet in that it's an intermittent Phantom bug.

In short, there is no bug in Ember CLI. You might propose we add a new feature and check for syntax errors before passing off to Phantom (we could be interested in you working on a feature like that!), but with inconsistent failures as reported by @wagenet the underlying problem exists in Phantom itself.

I'm closing this issue as it doesn't represent a bug in Ember CLI. Given your scenario is a situation with legitimately bad JS here are things you can do to help with this situation:
- Propose a quick check for syntax correctness as a last step in the Ember CLI build process in the Ember CLI RFCs repo and champion the implementation.
- Improve the error message inside of Phantom.

I'm happy to help you with the first option if you're interested. :smile:

**Comment by efx** [comment]
Yes, this problem seemed more fuzzy and wasn't sure where to put it.
In the above testcase, the stack trace is accurate but in my larger project the stack trace was entirely misleading. (sourcemap was off, but  that is a seperate issue).
I'll take the RFC route, thanks for suggesting that.
